kie-server-parent/kie-server-tests/kie-server-integ-tests-common/src/main/java/org/kie/server/integrationtests/shared/KieServerExecutor.java

```diff
@@ -121,7 +121,9 @@ public void stopKieServer() {
         if (server == null) {
             throw new RuntimeException("Kie execution server is already stopped!");
         }
-        kieServer.destroy();
+        if (kieServer != null) {
+            kieServer.destroy();
+        }
```

Comment:
To be reverted

Reply:
needed because kieserver may be null at this moment.
Instantiation of kieServer is in a different method than instantiation of server. 
If, by example,  

`private void addServerSingletonResources() {
        kieServer = new KieServerImpl();`
throws an exception, and the exception is catched by the calling code that then invoke

`public void stopKieServer() {`

at this moment kieServer is null, and without the null check 

`kieServer.destroy();`

throws another NPE.

If you navigate inside `new KieServerImpl();` you see that some exception may happen along the path.

/---/
I think that the lifecycle of those two objects (server/kieServer) should be more coupled to avoid inconsistence.